addon-resizer/nanny/main/pod_nanny.go

```diff
@@ -105,49 +127,29 @@ func main() {
 	// Use protobufs to improve performance.
 	config.ContentType = "application/vnd.kubernetes.protobuf"
 
-	k8s := nanny.NewKubernetesClient(*podNamespace, *deployment, *podName, *containerName, clientset, *useMetrics)
+	k8s := nanny.NewKubernetesClient(*podNamespace, *deployment, *podName, *containerName, clientset, *useMetrics, *runOnMaster)
 
 	nannyConfigurationFromFlags := &nannyconfigalpha.NannyConfiguration{
 		BaseCPU:       *baseCPU,
 		CPUPerNode:    *cpuPerResource,
 		BaseMemory:    *baseMemory,
 		MemoryPerNode: *memoryPerResource,
 	}
-	nannycfg, err := loadNannyConfiguration(*configDir, nannyConfigurationFromFlags)
-	if err != nil {
-		glog.Fatal(err)
-	}
-	glog.Infof("cpu: %s, extra_cpu: %s, memory: %s, extra_memory: %s", nannycfg.BaseCPU, nannycfg.CPUPerNode, nannycfg.BaseMemory, nannycfg.MemoryPerNode)
 
 	var resources []nanny.Resource
+	var nannyConfigUpdator nanny.NannyConfigUpdator
 
-	// Monitor only the resources specified.
-	if nannycfg.BaseCPU != nannyconfig.NoValue {
-		resources = append(resources, nanny.Resource{
-			Base:             resource.MustParse(nannycfg.BaseCPU),
-			ExtraPerResource: resource.MustParse(nannycfg.CPUPerNode),
-			Name:             "cpu",
-		})
-	}
-
-	if nannycfg.BaseMemory != nannyconfig.NoValue {
-		resources = append(resources, nanny.Resource{
-			Base:             resource.MustParse(nannycfg.BaseMemory),
-			ExtraPerResource: resource.MustParse(nannycfg.MemoryPerNode),
-			Name:             "memory",
-		})
-	}
-
-	if *baseStorage != nannyconfig.NoValue {
-		resources = append(resources, nanny.Resource{
-			Base:             resource.MustParse(*baseStorage),
-			ExtraPerResource: resource.MustParse(nannycfg.MemoryPerNode),
-			Name:             "storage",
-		})
+	if *runOnMaster {
+		nannyConfigUpdator = newNannyConfigUpdator(clientset, nannyConfigurationFromFlags, *podNamespace, *nannyConfigName, *baseStorage)
+	} else {
+        nannycfg, err := loadNannyConfiguration(*configDir, nannyConfigurationFromFlags)
+		if err != nil {
+			glog.Fatal(err)
+		}
+		glog.Infof("cpu: %s, extra_cpu: %s, memory: %s, extra_memory: %s", nannycfg.BaseCPU, nannycfg.CPUPerNode, nannycfg.BaseMemory, nannycfg.MemoryPerNode)
+		resources = updateResources(nannycfg, *baseStorage)
```

Comment:
I'm not fully understanding what you mean here. I want to keep original implementation more as is so that other users won't get too confused. We can do some refactor later if it's too complex. WDYT?

Reply:
I'm trying to minimize the amount of code paths. So my thought was, if you make a nannyConfigUpdater that only loads the config once, you don't need two completely different paths here, they would be more similar.

Reply:
I'm not sure if it would make it much cleaner if combining them together - the original code is reading config directly through file path while the new updater is to issue a Get call to apiserver. If we make a single updater then we will need to use some if statements to separate two different ways. Also the original config read is done in this main function, while the new updater is doing the calls in a for loop, which is also a bit different. And I still prefer to not touch the original config read so users can understand it easier that these are two different ways to fetch config. WDYT?

Reply:
Fair, let's keep it simple for now then.